test/bats/test.bats

```diff
@@ -72,41 +72,48 @@ BATS_TESTS_DIR=test/bats/tests
   assert_success
 
   cmd="kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/k8srequiredlabels.constraints.gatekeeper.sh"
-	waitForProcess "60" "1" "$cmd"
+	wait_for_process "60" "1" "$cmd"
 
   run kubectl apply -f ${BATS_TESTS_DIR}/constraints/all_ns_must_have_gatekeeper.yaml
   assert_success
 
   cmd="kubectl get k8srequiredlabels.constraints.gatekeeper.sh ns-must-have-gk -o yaml | grep enforced"
-  waitForProcess "60" "1" "$cmd"
+  wait_for_process "60" "1" "$cmd"
 
   run kubectl apply -f ${BATS_TESTS_DIR}/good/good_ns.yaml
   assert_success
 
   run kubectl apply -f ${BATS_TESTS_DIR}/bad/bad_ns.yaml
   assert_match 'denied the request' "$output"
   assert_failure
+}
 
-  # waiting for audit interval (default is 60s)
+# waiting for audit interval (default is 60s)
+@test "required labels audit test" {
   sleep 60
```

Comment:
can we drop the sleep 60 now that we're waiting for finalizers?

If it's still flaky, we can move the sleep to after the wait and reduce the duration.

My concern is that if we rely on guaranteed sleeps, the runtime for these tests will become very long.

Reply:
waiting for the finalizer still seems flaky for audit test unfortunately. moved audit test to the end, should give enough time for violations to be listed.

Reply:
oh, this is the audit test. Sorry, I was thinking it was the referential one.

Can we use something like the wait_for_process command that keeps checking for success and fails after a timeout?

It's fine to have a wait, I just don't want to wait a hard 60 seconds if we can finish earlier.

We can also speed up the audit interval, though that may have to be balanced against the performance of the test machine

Reply:
updated. issue was jsonpath was returning error code 0 even if it doesn't exist but looks like jq returns non-zero, added a wait before audit just in case. this should resolve this issue. keeping audit test at the end to reduce wait times.